java/src/main/java/ai/rapids/cudf/ArrowColumnBuilder.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *
+ *  Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringJoiner;
+
+/**
+ * Column builder from Arrow data. This builder takes in pointers to the Arrow off heap
+ * memory and allows efficient building of CUDF ColumnVectors from that arrow data.
+ * The caller can add multiple batches where each batch corresponds to Arrow data
+ * and those batches get concatenated together after being converted to CUDF
+ * ColumnVectors.
+ */
+public final class ArrowColumnBuilder implements AutoCloseable {
+    private DType type;
+    private ArrayList<Long> data = new ArrayList<>();
+    private ArrayList<Long> dataLength = new ArrayList<>();
+    private ArrayList<Long> validity = new ArrayList<>();
+    private ArrayList<Long> validityLength = new ArrayList<>();
+    private ArrayList<Long> offsets = new ArrayList<>();
+    private ArrayList<Long> offsetsLength = new ArrayList<>();
+    private ArrayList<Long> nullCount = new ArrayList<>();
+    private ArrayList<Long> rows = new ArrayList<>();
+    private int numBatches = 0;
+    private String colName;
+
+    public ArrowColumnBuilder(HostColumnVector.DataType type, String name) {
+      this.type = type.getType();
+      this.colName = name;
+    }
+
+    public void addBatch(long rows, long nullCount, long data, long dataLength, long valid,
+                         long validLength, long offsets, long offsetsLength) {
+      this.numBatches += 1;
+      this.rows.add(rows);
+      this.nullCount.add(nullCount);
```

Comment:
Do we really care about null count?  Would it be better to just let cudf calculate it if needed after we concat the data?

Reply:
so the Arrow spec says you don't need a validity vector when null count is 0. Since I'm just building the arrow array back up using this data, I thought it easier and safer to just build it back with the exact same values.  That way the c+  arrow array is the same as java array and CUDF can do what it wants from there.